src/connector/src/source/base.rs

```diff
@@ -477,6 +477,10 @@ impl SplitMetaData for SplitImpl {
             inner.update_with_offset(start_offset)
         )
     }
+
+    fn get_encoded_offset(&self) -> String {
```

Comment:
Given that this method is specific to dbz-based cdc source, I feel that it is confusing to add this method to the `SplitMetaData` trait and let all sources to impl this method. How about checking the type of SplitImpl in `need_commit_offset_to_upstream` and direct gets the cdc offset instead of introducing `get_encoded_offset` in the trait?